src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -426,6 +400,14 @@ public boolean onPortNumber(int portNumber) {
             throw new BindTransportException("Failed to bind to [" + port + "]", lastException.get());
         }
 
+        if (!DEFAULT_PROFILE.equals(name)) {
+            InetSocketAddress boundAddress = (InetSocketAddress) serverChannels.get(name).getLocalAddress();
+            int publishPort = settings.getAsInt("publish_port", boundAddress.getPort());
+            String publishHost = settings.get("publish_host", this.settings.get("transport.publish_host", this.settings.get("transport.host")));
+            InetSocketAddress publishAddress = createPublishAddress(publishHost, publishPort);
+            profileBoundAddresses.put(name, new BoundTransportAddress(new InetSocketTransportAddress(boundAddress), new InetSocketTransportAddress(publishAddress)));
```

Comment:
seems like some more code could be shared with the above block, unless I am missing something :) 
Also why do we only read `publish_port` here and not the other `transport.publish_port` etc. ?